internal/controlplane/common.go

```diff
@@ -49,3 +56,59 @@ func getNameFilterParam(name string) sql.NullString {
 		Valid:  name != "",
 	}
 }
+
+// getRemediationURLFromMetadata returns the "remediation URL". For now, this is
+// the URL link to the PR
+func getRemediationURLFromMetadata(data []byte, repoSlug string) (string, error) {
+	// If no data, it means no PR is tracked.
+	// So no error and we return an empty string.
+	if len(data) == 0 {
+		return "", nil
+	}
+	prData := &struct {
+		Number int `json:"pr_number"`
+	}{}
+
+	if err := json.Unmarshal(data, prData); err != nil {
+		return "", fmt.Errorf("unmarshalling pull request metadata: %w", err)
+	}
+
+	if repoSlug == "" && prData.Number == 0 {
```

Comment:
Do we need the second condition?  Is it ever valid to proceed with `repoSlug == ""`?

Reply:
This was actually a bug here and in the test. I've modified both URL functions to fail if no slug is set.